### PR TITLE
sys/phydat: rename phydat_scale_to_str -> phydat_prefix_from_scale

### PR DESCRIPTION
**Pull request by gebart**
### Contribution description

Correct some outdated documentation and make the name represent what the
function does today.

### Issues/PRs references

in #4668, `phydat_scale_to_str` was modified from its original form but the documentation was not updated

### CODE REVIEW

File: sys/include/phydat.h [hunk 1]
```diff
@@ -157,15 +156,17 @@ void phydat_dump(phydat_t *data, uint8_t dim);
 const char *phydat_unit_to_str(uint8_t unit);
 
 /**
- * @brief   Convert the given scale factor to a NULL terminated string
+ * @brief   Convert the given scale factor to an SI prefix
  *
  * The given scaling factor will be given as SI unit (e.g. M for Mega, u for
- * micro, etc) for obvious cases or in scientific notation (e.g. 2E11, 1E-22,
- * etc) otherwise.
+ * micro, etc), or `\0` otherwise.
```
Comment (aabadie) [inline sys/include/phydat.h]: Maybe rephrase a little the detailed descrption:
'The given scaling factor is returned in SI unit ...'

Comment (gebart) [reply sys/include/phydat.h]: addressed

**Comment by gebart** [comment]
squashed, waiting for Murdock
